jdbi/src/main/java/org/killbill/commons/jdbi/guice/DataSourceProvider.java

```diff
@@ -230,6 +243,57 @@ DataSource buildDataSource() {
         }
     }
 
+    private class UnpooledDataSourceBuilder {
+
+        DataSource buildDataSource() {
+            String dataSourceClassName = config.getDataSourceClassName();
+            if (Strings.isNullOrEmpty(dataSourceClassName)) {
+                throw new IllegalArgumentException("dataSourceClassName was not specified in the configuration and could not be inferred");
+            }
+
+            if (DatabaseType.MYSQL.equals(databaseType)) {
+                Reference ref = new Reference(dataSourceClassName);
+                ref.add(new StringRefAddr("user", config.getUsername()));
+                ref.add(new StringRefAddr("password", config.getPassword()));
+                ref.add(new StringRefAddr("explicitUrl", "true"));
+                ref.add(new StringRefAddr("url", config.getJdbcUrl()));
+                try {
+                    return (DataSource) new MysqlDataSourceFactory().getObjectInstance(ref, null, null, null);
```

Comment:
Re using HikariCP: I'm not sure how that would work.  [`ProxyFactory.getProxyConnection`](https://github.com/brettwooldridge/HikariCP/blob/HikariCP-2.3.7/hikaricp-common/src/main/java/com/zaxxer/hikari/proxy/ProxyFactory.java#L50) requires that one pass a [`HikariPool`](https://github.com/brettwooldridge/HikariCP/blob/HikariCP-2.3.7/hikaricp/src/main/java/com/zaxxer/hikari/pool/HikariPool.java) object as an argument.  `HikariPool` is a `final` class whose methods I can't override to behave the way I want them to.

Or are you saying I should use the Javassist library to [specify my own implementations for the `ProxyFactory` methods](https://github.com/brettwooldridge/HikariCP/blob/HikariCP-2.3.7/hikaricp/src/main/java/com/zaxxer/hikari/proxy/JavassistProxyFactory.java#L95-L120) at runtime?  I'm not clear on how that works, but I could look into it if you think that's a promising approach.

Re MySQL's `serverName` property:  since I provide the `url` property, am I not allowed to omit `serverName`?  I admit I haven't tested that yet.